packages/faceted-search/src/components/AddFacetPopover/AddFacetPopover.component.js

```diff
@@ -37,55 +69,189 @@ AddFacetRow.propTypes = {
+				defaultValue: 'Find a filter',
+			})}
+			onToggle={resetFilter}
+			onFilter={onFilter}
+			value={filterValue}
+			tabIndex={getTabIndex(isFocusable)}
+		/>
+	</RichLayout.Header>
+);
+
+AddFacetPopoverHeader.propTypes = {
+	category: PropTypes.string,
+	onCategoryChange: PropTypes.func.isRequired,
+	id: PropTypes.string.isRequired,
+	resetFilter: PropTypes.func.isRequired,
+	onFilter: PropTypes.func.isRequired,
+	filterValue: PropTypes.string.isRequired,
+	isFocusable: PropTypes.bool.isRequired,
+	t: PropTypes.func.isRequired,
+};
+
+const filterByLabel = label => row => {
+	const rowLabel = isString(row) ? row : row.properties.label;
+
+	return rowLabel.toLowerCase().includes(label);
 };
 
-const filterByLabel = label => badgeDefinition =>
-	badgeDefinition.properties.label.toLowerCase().includes(label);
+const sortByLabel = (rowA, rowB) => {
+	const labelA = isString(rowA) ? rowA : rowA.properties.label;
+	const labelB = isString(rowB) ? rowB : rowB.properties.label;
 
-const sortByLabel = (badgeDefinitionA, badgeDefinitionB) =>
-	badgeDefinitionA.properties.label.localeCompare(badgeDefinitionB.properties.label);
+	return labelA.localeCompare(labelB);
+};
 
 const AddFacetPopover = ({ badgesDefinitions = [], id, initialFilterValue, onClick, t }) => {
-	const [filterValue, setFilterValue] = useState(initialFilterValue || '');
+	const addFacetId = `${id}-add-facet-popover`;
+
+	const [category, setCategory] = useState(null);
+	const [filterValue, setFilterValue] = useState(initialFilterValue || '')
+
+	const categories = useMemo(
+		() =>
+			badgesDefinitions
+				.filter(badgeDefinition => !!badgeDefinition.metadata.category)
+				.map(badgeDefinition => badgeDefinition.metadata.category)
+				.filter((categoryName, index, arr) => arr.indexOf(categoryName) === index),
+		[badgesDefinitions],
+	);
+
+	const screensCount = categories.length + 1;
+	const [screensHeight, setScreensHeight] = useState(times(screensCount, () => 0));
+	const screensRef = useRef(times(screensCount, createRef));
+
+	useEffect(() => {
+		setScreensHeight(times(screensCount, index => screensRef.current[index].current.clientHeight));
+	}, []);
+
 	const onFilter = (_, value) => {
 		setFilterValue(value);
 	};
 	const resetFilter = () => setFilterValue('');
-	const badgesDefinitionsFaceted = useMemo(
+
+	const onCategoryChange = newCategory => {
+		setCategory(newCategory);
+		resetFilter();
+	};
+
+	const badgesWithoutCategory = useMemo(
 		() =>
-			badgesDefinitions.filter(filterByLabel(filterValue.toLowerCase().trim())).sort(sortByLabel),
-		[badgesDefinitions, filterValue],
+			badgesDefinitions.filter(badgeDefinition => !badgeDefinition.metadata.category),
+		[badgesDefinitions],
 	);
-	const addFacetId = `${id}-add-facet-popover`;
+
+	const screens = [
+		{
+			category: null,
+			rows: useMemo(
+				() =>
+					[...categories, ...badgesWithoutCategory]
+						.sort(sortByLabel)
+						.filter(filterByLabel(filterValue.toLowerCase().trim())),
```

Comment:
like @romainseb you can do the  filterValue.toLowerCase() in filterByLabel function